include/osmium/thread/pool.hpp

```diff
@@ -178,6 +182,10 @@ namespace osmium {
                 shutdown_all_workers();
             }
 
+            int num_threads() const {
+                return m_num_threads;
```

Comment:
BTW, `queue_size`  and `queue_empty` are not specified with `noexcept`. Is it intentional or the refactoring is in progress?